.github/workflows/autofix.yml

```diff
@@ -22,15 +22,15 @@ jobs:
         python-version: "3.11"
         cache: pip
     - run: pip install wheel
-    - run: pip install -r requirements-dev.txt
+    - run: pip install ".[dev]"
     - name: Gather a list of Python files in the pull request branch
       run: |
         (cd pr && git ls-files) | grep -E '\.(py|pyw)$' | sed s:^:pr/: | tee filelist.txt
     - run: python3 -m pycln --all --disable-all-dunder-policy $(cat filelist.txt)
     - run: python3 -m pyupgrade $(cat filelist.txt) --keep-runtime-typing --py38-plus --exit-zero-even-if-changed
     - run: python3 -m black $(cat filelist.txt)
-    - run: python3 -m isort $(cat filelist.txt)
+    - run: python3 -m ruff $(cat filelist.txt)
```

Comment:
Can we split ruff stuff into a separate PR? I want it to be easily revertible, especially the autofix stuff, because the autofix isn't exactly very easy to test before merging.